src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -500,6 +449,62 @@ Status RedisStoreClient::AsyncExists(const std::string &table_name,
   return Status::OK();
 }
 
+class Cleanup {
+ public:
+  Cleanup(std::function<void()> f) : f_(f) {}
+  ~Cleanup() { f_(); }
+
+ private:
+  std::function<void()> f_;
+};
+
+bool RedisDelKeyPrefixSync(const std::string &host,
+                           int32_t port,
+                           const std::string &password,
+                           bool use_ssl,
+                           const std::string &key_prefix) {
+  RedisClientOptions options(host, port, password, use_ssl);
+  auto cli = std::make_unique<RedisClient>(options);
+
+  instrumented_io_context io_service;
+
+  auto thread = std::make_unique<std::thread>([&]() {
+    boost::asio::io_service::work work(io_service);
+    io_service.run();
+  });
+
+  Cleanup _([&]() {
+    io_service.stop();
```

Comment:
this assumes io_service is GC'ed after Cleanup object. Is it true? maybe we should make io_service a shared pointer?

Reply:
same for thread

Reply:
yes this is safe. dtor call order is in reverse of ctor call order. so

io_service ctor
thread ctor
Cleanup ctor

// work

Cleanup dtor (stops io_service, joins thread)
thread dtor
io_service dtor

so we are safe.